app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,35 @@
 # app/controllers/posts_controller.rb
 class PostsController < ApplicationController
+  before_action :set_user, only: %i[index new create]
+
   def index
-    @user = User.find(params[:user_id])
     @posts = @user.posts.includes(:comments).paginate(page: params[:page], per_page: 6)
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = @user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] @DevKaranJ Great work adding a method to add a new post , however your method doesn't use the current_user method as per the project requirement and the post params is missing the title parameter, kindly consider fixing this error
![image](https://github.com/DevKaranJ/Blog/assets/61408860/b8feafa0-bb6e-422c-82c8-50dc7f07682f)